instance/tests/models/test_instance.py

```diff
@@ -503,12 +504,130 @@ def test_provision_mongo_again(self):
         self.check_mongo(instance)
 
 
+class EmailMixinInstanceTestCases(TestCase):
+    """
+    Test cases for EmailMixin
+    """
+
+    def _assert_called(self, mock, call_count=1):
+        """
+        Helper method - asserts that mock was called `call_count` times
+        """
+        self.assertTrue(mock.called)
+        self.assertEqual(len(mock.call_args_list), call_count)
+
+    def _get_call_arguments(self, mock, call_index=0):
+        """
+        Helper method - returns args and kwargs for `call_number`-th call
+        """
+        self.assertGreaterEqual(len(mock.call_args_list), call_index + 1)
+        args, unused_kwargs = mock.call_args_list[call_index]
+        return args
+
+    def _check_email_attachment(self, args, name, content_parts, mimetype=None):
+        """
+        Checks that `email.attach` arguments are as expected
+        """
+        attachment_name, attachment_content, attachment_mime = args
+
+        self.assertEqual(attachment_name, name)
+        for part in content_parts:
+            self.assertIn(part, attachment_content)
+
+        if mimetype:
+            self.assertEqual(attachment_mime, mimetype)
+
+    @patch("instance.models.mixins.utilities.EmailMultiAlternatives")
+    @override_settings(ADMINS=(("admin1", "admin1@localhost"),))
+    def test_provision_failed_email(self, mock_email_class):
+        """
+        Tests that provision_failed sends email when called from normal program flow
+        """
+        instance = OpenEdXInstanceFactory(name='test', sub_domain='test')
+        reason = "something went wrong"
+        mock_email_instance = Mock()
+        mock_email_class.return_value = mock_email_instance
```

Comment:
I think these tests could be simpler and easier to read if you use [the dummy outbox that django sets up](https://docs.djangoproject.com/en/1.8/topics/testing/tools/#email-services). Not a big deal, just a thought